modules/promotion/src/PromotionStorage.php

```diff
@@ -24,8 +24,27 @@ public function loadValid(OrderTypeInterface $order_type, StoreInterface $store)
     if (empty($result)) {
       return [];
     }
+
+    $usages_query = $this->database->select('commerce_promotion_usage', 'cpu');
```

Comment:
Feels wrong to query the DB table outside of the service. We have an API, let's use it.